.travis.yml

```diff
@@ -23,129 +31,62 @@ env:
-      env:
-        - SUITE=backend-integration
-        - REQUIREMENTS=prod
-        - EXTRAS=all,elasticsearch5,postgresql ES_VERSION=5.6.4
-
-      python:
-        - "3.6"
-
-      before_install:
-        - sudo service postgresql stop
-        - ./docker-inspire up -d
-        - curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python - --version 0.12.16
-        - source $HOME/.poetry/env
-        - travis_retry pip install twine wheel coveralls
-      install:
-        - ./scripts/bootstrap --ci
-
-      before_script:
-        - ./docker/wait-for-services.sh
-
-      script:
-        - poetry run py.test tests/integration
-
-      after_script:
-        - docker-compose kill -s SIGTERM
-        - docker-compose rm -f
-
-      after_success:
-        - coveralls
-
-      after_fail:
-        - poetry show
-
-    - language: python
-      sudo: true
-      dist: trusty
-      env:
-        - SUITE=backend-unit
-        - REQUIREMENTS=prod
-      python:
-        - "3.6"
-
-      before_install:
-        - curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python - --version 0.12.16
-        - source $HOME/.poetry/env
-        - travis_retry pip install twine wheel coveralls
-
-      install:
-        - ./scripts/bootstrap --ci
-
-      script:
-        - ./run-code-checks.sh
-        - poetry run py.test tests/unit
-
-      after_success:
-        - coveralls
-
-      after_fail:
-        - poetry run pip freeze
-
-    - language: python
-      sudo: true
-      dist: trusty
-      env:
-        - SUITE=backend-integration-async
-        - REQUIREMENTS=prod
-        - EXTRAS=all,elasticsearch5,postgresql ES_VERSION=5.6.4
-
-      # XXX: needs work around for 3.7
-      python:
-        - "3.6"
-
-      before_install:
-        - sudo service postgresql stop
-        - ./docker-inspire up -d
-        - curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python - --version 0.12.16
-        - source $HOME/.poetry/env
-        - travis_retry pip install twine wheel coveralls
-
-      install:
-        - ./scripts/bootstrap --ci
-
-      before_script:
-        - ./docker/wait-for-services.sh
-
-      script:
-        - poetry run py.test tests/integration-async
-
-      after_script:
-        - docker-compose kill -s SIGTERM
-        - docker-compose rm -f
-
-      after_success:
-        - coveralls
-
-      after_fail:
-        - poetry run pip freeze
-
-    - language: node_js
-      sudo: required
-      env:
-        - SUITE=ui-all
-      node_js:
-        - 10.14.0
-      script: cd ui && yarn && yarn test
+    - env: SUITE=integration RUN_DOCKER=true
+    - env: SUITE=unit RUN_DOCKER=false
+    - env: SUITE=integration-async RUN_DOCKER=true
+    - env: SUITE=ui RUN_DOCKER=false
 
 deploy:
+  skip_cleanup: true
   provider: script
   script: ./build-docker-images.sh
   on:
     branch: master
+    condition: $SUITE = integration
```

Comment:
this is awkward, to deploy UI on `integration` suite. What if UI suite fails? this will stil do the deployment.

Reply:
yeah, might be better to deploy backend on `integration` and ui  on `ui` if not too complex